metrics/config.go

```diff
@@ -341,6 +353,25 @@ func prometheusPort() (int, error) {
 	return int(pp), nil
 }
 
+// prometheusHost returns the host configured via the environment
+// for the Prometheus metrics exporter if it's set, a default value otherwise.
+// Validation is performed via parsing the host value as part of a url.
+func prometheusHost() (*string, error) {
```

Comment:
I don't think returning string pointer is a common practice in Golang or eventing. For this particular case, it seems to me that you don't need a nil host value at all. For performance, RVO will optimize it out anyway.

Reply:
ok.

Reply:
Yes, that's the common case of using *string, but in your case you already have error return value, so when error is not nil, string should be ignored. That's why I said "in this particular case".

Reply:
Returning *string is useful if we need to discern between empty and nil and is used a lot in APIs around K8s.
Here it is probably a tossup, so should not matter.